### PR TITLE
Add Monitoring Variables and Fix Socket Error

### LINKED ISSUE #16: CTB as HSI giving (faulty?) error on stop of run
Seeing following error on stop of run from CTB when running as HSI

```
 CTB Hardware Communication Error: Socket closed: End of file
```

Can restart a run with no problem, so possibly this is not such a fatal issue, but some out-of-order operation happening int he stop transition.

### PR DESCRIPTION
**Pull request by jsensenig**
Adding monitoring metrics into the CTBmodule and fixing a prematurely closing socket at end of run.

Monitoring variables and associated schema added to monitor the HLT, LLT and timestamp word counts. Only count the configured triggers.

Fix error where the CTB gives an end of file error from the receiver bugger when stopping run. Add a short sleep in do_stop() to allow receiving loop to end before CTB hardware is stopped.
resolves https://github.com/DUNE-DAQ/ehn1-operations-issues/issues/16

### CODE REVIEW

**Comment by jsensenig** [comment]
These changes have been successfully tested with `daqsystemtest: minimal_system_quick_test.py` and detector run 24632.

File: plugins/CTBModule.cpp [hunk 1]
```diff
@@ -128,6 +154,9 @@ CTBModule::do_start(const nlohmann::json& startobj)
 
   TLOG_DEBUG(TLVL_ENTER_EXIT_METHODS) << get_name() << ": Entering do_start() method";
 
+  // Set this to false early so it doesn't interfere with the start
+  m_stop_requested.store(false);
+
```
Comment (mroda88) [inline plugins/CTBModule.cpp:159]: We should also reset `m_total_hlt_counter` to 0 here.

**Review by mroda88 — CHANGES_REQUESTED** [review]
Hi Jon, this is good. But the absolute counters should also be reset every time we call for start. That is because we might restart a run without reconfiguring. 
Can you make the change?

File: plugins/CTBModule.cpp (hunk 1 above)
Comment (jsensenig) [reply plugins/CTBModule.cpp:159]: Hey Marco, good point, I added the counter reset.

**Review by mroda88 — APPROVED** [review]
All good for me and plots on grafana are coming through correctly
Screenshot from 2024-03-26 12-18-13

**Pull request merged**